app/src/main/java/ca/marshallasch/veil/PostListAdapter.java

```diff
@@ -284,11 +305,20 @@ protected FilterResults performFiltering(CharSequence charSequence)
             }
 
             @Override
-            protected void publishResults(CharSequence charSequence, FilterResults filterResults)
-            {
+            protected void publishResults(CharSequence charSequence, FilterResults filterResults) {
                 postsFiltered = (ArrayList<DhtProto.Post>) filterResults.values;
                 notifyDataSetChanged();
             }
         };
     }
+
+    private void updatePostStatus(ViewHolder1 viewHolder, DhtProto.Post post){
+        int postStatus = DataStore.getInstance(activity).getPostStatus(post.getUuid());
+
+        if(postStatus == KnownPostsContract.POST_NORMAL){
```

Comment:
This can be reduced to being 3 lines. by using a conditional set for a resource then only set the image once after the resource to use is updated.

Reply:
resovled